### PR TITLE
[Feat]: Add GQA Attention Sink backward bhsd example

### PR DESCRIPTION
**Pull request by wch0810**
## Description

Add GQA + Attention Sink Flash Attention (Forward + Backward, BHSD layout) for Ascend NPU, implemented in `example_gqa_sink_bwd_bhsd.py` based on TileLang + AscendC PTO.

5 kernels (forward, backward preprocess, backward main 5‑GEMM, postprocess, dsink) + autograd wrapper. Expert mode for forward/backward (dynamic loop bounds), combineCV pass for simple kernels. Layered test suite (37 cases) + performance benchmark (`do_bench` + `msprof op`).

Tested on Ascend NPU, fp16, B=1, H=64, N=4096, D=128, groups=8, window=128. Precision verified against PyTorch golden (fwd atol=5e‑3, bwd atol=1e‑2, 37/37 PASS).

## Golden Config (B=1, H=64, N=4096, D=128, groups=8, window=128, fp16)

| Kernel                              | Latency (ms) | TFLOPS | Max Diff |
| ----------------------------------- | ------------ | ------ | -------- |
| TileLang Forward                    | 5.64         | 3.00   | 1.95e‑03 |
| TileLang Backward                   | 5.51         | 7.67   | 7.81e‑03 |
| TileLang Fwd+Bwd+Post+Dsink (e2e)   | 14.49        | 4.09   | —        |
| PyTorch Fwd+Bwd (e2e)               | 200.65       | 0.30   | —        |
| Baseline Backward (same cfg)        | 14.29        | 2.96   | —        |

- NPU e2e **13.85× faster than PyTorch**. NPU Backward **2.59× faster than Baseline**.

## Notes

- **Dynamic loop bounds**: `loop_st`/`loop_ed` skip fully‑invalid KV blocks (window=128 → 97% compute skipped, 20× speedup). Applied to both forward and backward. This optimization requires Expert mode (manual `T.Scope`/sync) because combineCV pass requires constant loop extents.
- **combineCV**: simple kernels (preprocess/postprocess/dsink) use combineCV pass (no `T.Scope`/`barrier_all`). Forward/backward retain Expert mode (dynamic loop bounds incompatible with combineCV, verified by experiment — compile fails with `"only works with constant loop sizes, but got cid % 2 + 1"`).
- **lse/delta broadcast**: `T.tile.broadcast(axis=1)` replaces per‑row for loops (19% backward speedup).
- **Autograd**: `attention(q, k, v, sinks, window_size, groups)` — `O.backward(dO)` triggers all 5 backward kernels automatically.
- **dK/dV atomic_add**: zeroed before each call. dQ uses `init=True` overwrite.
- **Block sizes**: block_M=64, block_N=64. `dim_qk_padded = ceildiv(D, 128) * 128`.
- **Performance**: `do_bench` is CI default (stable, no CANN dependency); `--level msprof` available for kernel‑level analysis (L2 hit rate, pipeline utilization, etc., local‑only, skipped in CI).
- **Limitations**: D ≤ 128; `seq_len` % 64 == 0; `window_size` % 64 == 0; fp16 only.

### CODE REVIEW

File: examples/attention_sink/example_gqa_sink_bwd_bhsd/test_gqa_sink_bwd_bhsd.py [hunk 1]
```diff
@@ -0,0 +1,940 @@
...
+        O.backward(dO)
+
+        # Golden backward
+        dQ_ref, dK_ref, dV_ref, dSinks_ref = ref_bwd(Q.detach(), K.detach(), V.detach(), sinks.detach(), dO, window_size, groups)
+
+        # Compare gradients
+        torch.testing.assert_close(Q.grad.cpu(), dQ_ref.cpu(), rtol=RTOL, atol=ATOL)
+        torch.testing.assert_close(K.grad.cpu(), dK_ref.cpu(), rtol=RTOL, atol=ATOL)
+        torch.testing.assert_close(V.grad.cpu(), dV_ref.cpu(), rtol=RTOL, atol=ATOL)
+        torch.testing.assert_close(sinks.grad.cpu(), dSinks_ref.cpu(), rtol=RTOL, atol=ATOL)
```
Comment (hedi515) [inline examples/attention_sink/example_gqa_sink_bwd_bhsd/test_gqa_sink_bwd_bhsd.py]: 精度标准不对，参考.agents/skills/tilelang-op-test-design/references/precision-standard.md

File: examples/attention_sink/example_gqa_sink_bwd_bhsd/example_gqa_sink_bwd_bhsd.py [hunk 2]
```diff
@@ -0,0 +1,1067 @@
...
+            v_row = vid * hm  # V-scope row offset within block_M
+            q_row = bx * block_M  # Q block start row in GM
+
+            with T.Scope("C"):
+                T.copy(Q[bz, by, q_row : q_row + block_M, :], q_l1)
+                T.barrier_all()
+                for k in T.serial(loop_st, loop_ed):
+                    kv = k * block_N
+                    T.copy(K[bz, kv_by, kv : kv + block_N, :], k_l1)
+                    T.barrier_all()
```
Comment (hedi515) [inline examples/attention_sink/example_gqa_sink_bwd_bhsd/example_gqa_sink_bwd_bhsd.py:225]: 这么多barrier_all，这种写法性能好吗
